extensions/wikia/CreateNewWiki/tasks/TaskHelper.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Wikia\CreateNewWiki\Tasks;
+
+class TimeCounter {
+
+	private $startTime;
+
+	public function __construct() {
+		$this->startTime = microtime( true );
+	}
+
+	public function getTimePassed() {
+		return microtime( true ) - $this->startTime;
+	}
+
+	public function getContext( $context = null ) {
+		$timeContext = [ 'took' => $this->getTimePassed() ];
+		if ( !empty( $context ) ) {
+			return array_merge( $context, $timeContext );
+		}  else {
+			return $timeContext;
+		}
+	}
+}
+
+class TaskHelper {
+
+	/**
+	 * Wait for shared DB and the current DB cluster slaves
+	 *
+	 * @param TaskContext $taskContext
+	 * @param string $functionName
+	 * @see PLATFORM-1219
+	 */
```

Comment:
Yes, apparently CNW was failing and wait for slaves was added to mitigate that problem. I think the mention here is just to provide additional context, it does not suggest that we need to do anything about this function when the ticket is resolved, so the comment is fine as is.